compiler/tools/fpp-to-cpp/test/component/base/ActiveStateMachinesComponentAc.ref.hpp

```diff
@@ -0,0 +1,103 @@
+// ======================================================================
+// \title  ActiveStateMachinesComponentAc.hpp
+// \author Generated by fpp-to-cpp
+// \brief  hpp file for ActiveStateMachines component base class
+// ======================================================================
+
+#ifndef ActiveStateMachinesComponentAc_HPP
+#define ActiveStateMachinesComponentAc_HPP
+
+#include "ActiveStateMachines_S1.hpp"
+#include "ActiveStateMachines_S2.hpp"
+#include "FpConfig.hpp"
+#include "Fw/Comp/ActiveComponentBase.hpp"
+#include "Fw/Port/InputSerializePort.hpp"
+#include "Fw/Port/OutputSerializePort.hpp"
+#include "Fw/Types/SMEventsSerializableAc.hpp"
+
+//! \class ActiveStateMachinesComponentBase
+//! \brief Auto-generated base for ActiveStateMachines component
+//!
+//! An active component with state machines
+class ActiveStateMachinesComponentBase :
+  public Fw::ActiveComponentBase, public ActiveStateMachines_S1_Interface, public ActiveStateMachines_S2_Interface
+{
+
+  PROTECTED:
+
+    // ----------------------------------------------------------------------
+    // State machine enumeration
+    // ----------------------------------------------------------------------
+    enum SmId {
+      STATE_MACHINE_SM1,
+      STATE_MACHINE_SM2,
+      STATE_MACHINE_SM3,
+      STATE_MACHINE_SM4,
+    };
+
+    // ----------------------------------------------------------------------
+    // Friend classes
+    // ----------------------------------------------------------------------
+
+    //! Friend class for white-box testing
+    friend class ActiveStateMachinesComponentBaseFriend;
+
+  public:
+
+    // ----------------------------------------------------------------------
+    // Component initialization
+    // ----------------------------------------------------------------------
+
+    //! Initialize ActiveStateMachinesComponentBase object
+    void init(
+        FwQueueSizeType queueDepth, //!< The queue depth
+        FwEnumStoreType instance = 0 //!< The instance number
+    );
+
+  PROTECTED:
+
+    // ----------------------------------------------------------------------
+    // Component construction and destruction
+    // ----------------------------------------------------------------------
+
+    //! Construct ActiveStateMachinesComponentBase object
+    ActiveStateMachinesComponentBase(
+        const char* compName = "" //!< The component name
+    );
+
+    //! Destroy ActiveStateMachinesComponentBase object
+    virtual ~ActiveStateMachinesComponentBase();
+
+  PROTECTED:
+
+    // ----------------------------------------------------------------------
+    // State machine function to push events to the input queue
+    // ----------------------------------------------------------------------
+
+    //! State machine base-class function for sendEvents
```

Comment:
We need to use the term "signal" instead of the term "event," per the recent changes in the spec.

Reply:
All references to Events have been changed to Signals

Reply:
I'm still seeing references to state machine events:
```
[I] bocchino@MT-504857 ~/J/T/f/g/f/c/t/f/t/c/base (statemachine-code)> grep -i Event ActiveStateMachinesComponentAc.{hpp,cpp}
ActiveStateMachinesComponentAc.hpp:#include "Fw/Types/SMEventsSerializableAc.hpp"
ActiveStateMachinesComponentAc.hpp:    // State machine function to push events to the input queue
ActiveStateMachinesComponentAc.hpp:    //! State machine base-class function for sendEvents
ActiveStateMachinesComponentAc.hpp:        const Fw::SMEvents& ev //!< The state machine event
ActiveStateMachinesComponentAc.cpp:    STATEMACHINE_SENDEVENTS,
ActiveStateMachinesComponentAc.cpp:    // Size of statemachine sendEvents
ActiveStateMachinesComponentAc.cpp:    BYTE sendEventsStatemachineSize[
ActiveStateMachinesComponentAc.cpp:      Fw::SMEvents::SERIALIZED_SIZE
ActiveStateMachinesComponentAc.cpp:// State machine function to push events to the input queue
ActiveStateMachinesComponentAc.cpp:  stateMachineInvoke(const Fw::SMEvents& ev)
ActiveStateMachinesComponentAc.cpp:  _status = msg.serialize(static_cast<FwEnumStoreType>(STATEMACHINE_SENDEVENTS));
ActiveStateMachinesComponentAc.cpp:    // Handle state machine events 
ActiveStateMachinesComponentAc.cpp:    case STATEMACHINE_SENDEVENTS: {
ActiveStateMachinesComponentAc.cpp:      Fw::SMEvents ev;
ActiveStateMachinesComponentAc.cpp:      // Update the state machine with the event
```
I think all these need to be changed to signals.